### PR TITLE
docs: fleet setup scripts

### PR DESCRIPTION
**Pull request by britaniar**
### Description of your changes

Hub cluster setup and member cluster joining scripts.

Fixes #

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

**Comment by ryanzhang-oss** [comment]
Can you please add a READM.md with step-by-step instruction on how to run those script?

File: hack/setup/README.md [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+    --set logVerbosity=2 \
+    --set namespace=fleet-system \
+    --set enableWebhook=false \
+    --set webhookClientConnectionType=service \
+    --set enableV1Alpha1APIs=false \
+    --set enableV1Beta1APIs=true \
+    --set resources.limits.cpu=4 \
+    --set resources.limits.memory=4Gi \
+    --set concurrentClusterPlacementSyncs=10 \
+    --set ConcurrentRolloutSyncs=10 \
```
Comment (ryanzhang-oss) [inline hack/setup/README.md:128]: what is this?

Comment (britaniar) [reply hack/setup/README.md]: That was one of the parameters we added at the beginning but then ended up taking off. Forgot to take it off of the wiki I created that I copied this over from.

File: hack/setup/README.md [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+> * `base64`
+
+## Create a hub cluster from a cluster 
+For your convenience, Fleet provides a script that can automate the process of creating a hub cluster. To use script,
+run the commands bellow:
+```sh
+# Replace the value of HUB_CLUSTER_CONTEXT with the name of the kubeconfig context you use for
+# accessing your hub cluster.
+export HUB_CLUSTER_CONTEXT=<YOUR-HUB-CLUSTER-CONTEXT>
+# Replace the value of HUB_CLUSTER_ADDRESS with the address of your hub cluster API server.
```
Comment (ryanzhang-oss) [inline hack/setup/README.md:26]: how do we know the hub api server address before we create the hub cluster?

File: hack/setup/README.md [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+For your convenience, Fleet provides a script that can automate the process of joining a cluster
+onto a hub cluster. To use the script, run the commands below after creating needed AKS clusters:
+
+In `joinMC.sh` file line 4, replace current values of `MC_NAMES` with the names of the AKS cluster you want to join.
+```sh
+# Replace the value of HUB_CLUSTER_CONTEXT with the name of the kubeconfig context you use for
+# accessing your hub cluster.
+export HUB_CLUSTER_CONTEXT=YOUR-HUB-CLUSTER-CONTEXT
+# Replace the value of HUB_CLUSTER_ADDRESS with the address of your hub cluster API server.
+export HUB_CLUSTER_ADDRESS=YOUR-HUB-CLUSTER-ADDRESS
```
Comment (ryanzhang-oss) [inline hack/setup/README.md:93]: is this a duplicate of line 23 to 27?

File: hack/setup/README.md [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+
+# Replace <MEMBER-CLUSTER-NAME1> and <MEMBER-CLUSTER-NAME-2> with a list of cluster names (separated by a space as a string)
+# that you would like to join the fleet as member clusters. Their context will be used to access the cluster.
+# Ex.: export MC_NAMES_STR="member member2"
+export MC_NAMES_STR="<MEMBER-CLUSTER-NAME-1> <MEMBER-CLUSTER-NAME-2>"
```
Comment (ryanzhang-oss) [inline hack/setup/README.md:98]: can we  move those as parameters to the script?

File: hack/setup/README.md [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+# Replace <MEMBER-CLUSTER-NAME1> and <MEMBER-CLUSTER-NAME-2> with a list of cluster names (separated by a space as a string)
+# that you would like to join the fleet as member clusters. Their context will be used to access the cluster.
+# Ex.: export MC_NAMES_STR="member member2"
+export MC_NAMES_STR="<MEMBER-CLUSTER-NAME-1> <MEMBER-CLUSTER-NAME-2>" 
+
+# Clone the Fleet GitHub repository, if necessary.
+git clone https://github.com/Azure/fleet.git
+# Go into fleet directory.
+cd fleet
```
Comment (ryanzhang-oss) [inline hack/setup/README.md:103]: is this a duplicate of line 29-32?

File: hack/setup/README.md (hunk 2 above)
Comment (britaniar) [reply hack/setup/README.md]: Added a note that says to clusters are a prerequisite before running the script and has the azure link to create clusters.

**Review by ryanzhang-oss — COMMENTED** [review]
can you move this under Azure folder?

File: hack/Azure/setup/README.md [hunk 6]
```diff
@@ -0,0 +1,117 @@
+# Setting up a Fleet
+
+This how-to guide discusses how to create a fleet, specifically:
```
Comment (ryanzhang-oss) [inline hack/Azure/setup/README.md:3]:
```suggestion
This how-to guide describes how to create a fleet using Azure Kubernetes Service, specifically:

```

File: hack/Azure/setup/README.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
+# Setting up a Fleet
+
+This how-to guide discusses how to create a fleet, specifically:
+
+* how to create a hub cluster; and
```
Comment (ryanzhang-oss) [inline hack/Azure/setup/README.md:5]:
```suggestion
* how to create an AKS cluster as the  hub cluster; and

```

File: hack/Azure/setup/README.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
+# Setting up a Fleet
+
+This how-to guide discusses how to create a fleet, specifically:
+
+* how to create a hub cluster; and
+* how to join clusters to hub cluster; and
```
Comment (ryanzhang-oss) [inline hack/Azure/setup/README.md:6]:
```suggestion
* how to join any k8s clusters to the AKS hub cluster; and

```

File: hack/Azure/setup/README.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
+# Setting up a Fleet
+
+This how-to guide discusses how to create a fleet, specifically:
+
+* how to create a hub cluster; and
+* how to join clusters to hub cluster; and
+* how to add labels to a member cluster
```
Comment (ryanzhang-oss) [inline hack/Azure/setup/README.md:7]:
```suggestion
* how to add labels to a member cluster representation on the hub cluster

```

**Review by ryanzhang-oss — APPROVED** [review]

**Pull request merged**